typescript/sdk/src/deploy/middleware/LiquidityLayerApp.ts

```diff
@@ -75,6 +73,44 @@ export class LiquidityLayerApp<
     return response.result.map((_: any) => _.transactionHash).flat();
   }
 
+  async fetchPortalBridgeTransactions(chain: Chain): Promise<string[]> {
+    const cc = this.multiProvider.getChainConnection(chain);
+    const params = new URLSearchParams({
+      module: 'logs',
+      action: 'getLogs',
+      address: this.getContracts(chain).portalAdapter!.address,
+      topic0: PortalBridgedTokenTopic,
+    });
+    const req = await fetch(`${cc.getApiUrl()}?${params}`);
+    const response = await req.json();
+
+    return response.result.map((_: any) => _.transactionHash).flat();
+  }
+
+  async parsePortalMessages(
+    chain: Chain,
+    txHash: string,
+  ): Promise<PortalBridgeMessage<Chain>[]> {
+    const connection = this.multiProvider.getChainConnection(chain);
+    const receipt = await connection.provider.getTransactionReceipt(txHash);
+    const matchingLogs = receipt.logs
+      .map((_) => {
+        try {
+          return [PortalAdapterInterface.parseLog(_)];
+        } catch {
+          return [];
+        }
+      })
+      .flat();
+    if (matchingLogs.length == 0) return [];
+
+    const event = matchingLogs.find((_) => _!.name === 'BridgedToken')!;
+    const portalSequence = event.args.portalSequence.toNumber();
+    const nonce = event.args.nonce.toNumber();
+    const destination = DomainIdToChainName[event.args.destination];
+    // @ts-ignore
```

Comment:
curious why this was needed

Reply:
The compiler cannot infer whether `destination` is within the list of `Chain` set as a type guard